cupy/lib/polynomial.pyx

```diff
@@ -4,6 +4,47 @@ import cupy
 from cupy.core.core cimport ndarray
 
 
+def poly(seq):
+    """Computes the coefficients of a polynomial with the given roots sequence.
+
+    Args:
+        seq (cupy.ndarray): a sequence of polynomial roots.
+
+    Returns:
+        cupy.ndarray: polynomial coefficients from highest to lowest degree.
+
+    .. warning::
+
+        This function doesn't support general 2d square arrays currently.
+        Only complex Hermitian and real symmetric 2d arrays are allowed.
+
+    .. seealso:: :func:`numpy.poly`
+
+    """
+    if seq.ndim == 2 and seq.shape[0] == seq.shape[1] and seq.shape[0] != 0:
+        if cupy.array_equal(seq, seq.conj().T):
+            seq = cupy.linalg.eigvalsh(seq)
+        else:
+            raise NotImplementedError('Only complex Hermitian and real '
+                                      'symmetric 2d arrays are supported '
+                                      'currently')
+    elif seq.ndim == 1:
+        seq = seq.astype(cupy.mintypecode(seq.dtype.char), copy=False)
+    elif seq.ndim == 0:
+        raise TypeError('Input must be 1d or non-empty square 2d array.')
+    else:
+        raise ValueError('Input must be 1d or non-empty square 2d array.')
+
+    if seq.size == 0:
+        return 1.0
+
+    out = cupy.ones((1,), seq.dtype)
+    seq = cupy.column_stack((cupy.ones(seq.size, seq.dtype), -seq))
+    for zero in seq:
+        out = cupy.convolve(out, zero, 'full')
```

Comment:
Please parallelize this loop. We can achieve this computation with O(log n) time FFT call.

Reply:
The best algorithm I know for coefficients computation using FFT is a recursive O(n log^2 n) time one. Is that what you mean or do you have a better approach?

Reply:
Yes.

Reply:
```
def _fft_poly(seq):
    n = seq.shape[0]
    if n == 1:
        return seq[0]
    if n == 2:
        return cupy.math.misc._fft_convolve(seq[0], seq[1], 'full')
    return signal.convolve(_fft_poly(seq[:n / 2]), _fft_poly(seq[n / 2:]), 'full', 'direct')
```
I'm not really sure if that's what you meant or not. If so, I want to clarify that
1. complex dtype give different results in comparison with numpy's
2. Only `direct` method is currently supported for `cupyx.scipy.signal.convolve` that's why I have to specify the method.

Reply:
No, it requires O(n) time FFT kernel call.